Datagrid/ProxyQueryInterface.php

```diff
@@ -15,6 +15,10 @@
  * Used by the Datagrid to build the query.
  *
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * @mixin \Doctrine\ODM\MongoDB\Query\Builder
+ * @mixin \Doctrine\ODM\PHPCR\Query\Builder\QueryBuilder
+ * @mixin \Doctrine\ORM\QueryBuilder
```

Comment:
What is this? Admin shouldn't know about persistence layers, should it?

Reply:
This will add autocompletion in PhpStorm

Reply:
I don't use PS, so this is not going to be very convincing I'm afraid :P And it looks wrong to have namespaces of uninstalled classes in the code.

Reply:
I use PHPStorm, but I agree with @greg0ire for this one.

BTW, it will be quite confusing. If you receive an interface, you should use the method of the interface or do `instanceof` check (and then you will have completion).

Reply:
Thought this would help, but I removed the commit for now